File: Application/Common/Filters/DetailFilter.cs
using DataAnalysis.Application.Common.Enums;

namespace DataAnalysis.Application.Common.Filters;

public sealed class DetailFilter
{
    public IReadOnlyList<InsuredType>    InsuredTypes    { get; init; } = [];
    public IReadOnlyList<BusinessSource> BusinessSources { get; init; } = [];
    public IReadOnlyList<VehicleType>    VehicleTypes    { get; init; } = [];
    public IReadOnlyList<string>         ProductCodes    { get; init; } = [];

    public bool IsEmpty =>
        InsuredTypes.Count == 0 && BusinessSources.Count == 0 &&
        VehicleTypes.Count == 0 && ProductCodes.Count == 0;
}

File: Client/src/utils/detailFilter.js
export const EMPTY_FILTER = {
  insuredTypes: [],
  businessSources: [],
  vehicleTypes: [],
  productCodes: [],
}

export function buildFilterParams(productGroup, filter) {
  const p = new URLSearchParams()
  if (productGroup) p.append('productGroup', productGroup)
  if (filter) {
    ;(filter.insuredTypes    || []).forEach(v => p.append('insuredType', v))
    ;(filter.businessSources || []).forEach(v => p.append('businessSource', v))
    ;(filter.vehicleTypes    || []).forEach(v => p.append('vehicleType', v))
    ;(filter.productCodes    || []).forEach(v => p.append('product', v))
  }
  return p
}

export function isFilterActive(filter) {
  if (!filter) return false
  return (
    (filter.insuredTypes?.length    || 0) +
    (filter.businessSources?.length || 0) +
    (filter.vehicleTypes?.length    || 0) +
    (filter.productCodes?.length    || 0)
  ) > 0
}

export const stripBusinessSource = (filter) => ({ ...filter, businessSources: [] })